planning/obstacle_avoidance_planner/src/utils/cv_utils.cpp

```diff
@@ -407,42 +410,77 @@ boost::optional<Edges> getEdges(
 
 namespace cv_drivable_area_utils
 {
+
 bool isOutsideDrivableAreaFromRectangleFootprint(
   const autoware_auto_planning_msgs::msg::TrajectoryPoint & traj_point,
   const cv::Mat & road_clearance_map, const nav_msgs::msg::MapMetaData & map_info,
-  const VehicleParam & vehicle_param)
+  const VehicleParam & vehicle_param, const nav_msgs::msg::OccupancyGrid & drivable_area_grid,
+  const bool grid_iter)
 {
   const double base_to_right = (vehicle_param.wheel_tread / 2.0) + vehicle_param.right_overhang;
   const double base_to_left = (vehicle_param.wheel_tread / 2.0) + vehicle_param.left_overhang;
 
+  const double half_width = vehicle_param.width / 2.0;
   const double base_to_front = vehicle_param.length - vehicle_param.rear_overhang;
   const double base_to_rear = vehicle_param.rear_overhang;
 
+  constexpr double epsilon = 1e-8;
+
   const auto top_left_pos =
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, base_to_front, -base_to_left, 0.0)
       .position;
+  tier4_autoware_utils::calcOffsetPose(traj_point.pose, base_to_front, -half_width, 0.0).position;
   const auto top_right_pos =
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, base_to_front, base_to_right, 0.0)
       .position;
+  tier4_autoware_utils::calcOffsetPose(traj_point.pose, base_to_front, half_width, 0.0).position;
   const auto bottom_right_pos =
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, base_to_right, 0.0)
       .position;
+  tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, half_width, 0.0).position;
   const auto bottom_left_pos =
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, -base_to_left, 0.0)
       .position;
+  tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, -half_width, 0.0).position;
+
+  if (grid_iter) {
+    grid_map::GridMap grid_map;
+    grid_map::GridMapRosConverter::fromOccupancyGrid(drivable_area_grid, "layer", grid_map);
+    const grid_map::Matrix & grid_data = grid_map["layer"];
+
+    std::vector<grid_map::Position> positions = {
+      {top_left_pos.x, top_left_pos.y},
+      {top_right_pos.x, top_right_pos.y},
+      {bottom_right_pos.x, bottom_right_pos.y},
+      {bottom_left_pos.x, bottom_left_pos.y},
+    };
+
+    for (auto i = 0lu; i + 1 < positions.size(); ++i) {
+      const grid_map::Position start(positions[i]);
+      const grid_map::Position end(positions[i + 1]);
+      if (grid_map.isInside(start) || grid_map.isInside(end)) {
```

Comment:
I just found I made a mistake and the `LineIterator` crashes if one of the points is outside the map so this condition should be changed.
```suggestion
      if (grid_map.isInside(start) && grid_map.isInside(end)) {
```
This means that the footprints on the edges of the maps will not be checked.